crud/src/main/java/com/redhat/lightblue/mediator/Mediator.java

```diff
@@ -98,13 +99,13 @@
 
     public final Metadata metadata;
     public final Factory factory;
-
-    public Mediator(Metadata md,
-                    Factory factory) {
+    public final RequestMetrics metrics;
+   
+    public Mediator(Metadata md, Factory factory, RequestMetrics metrics) {
```

Comment:
This is not good. Why is metrics being passed in ctor? Ideally, metrics should be in the Factory, or LightblueConfiguration, and not passed like this. 

Does this metrics class work only at the mediator level? If so, a cleaner approach would be to remove all metrics related code from the mediator, and write a MediatorWithMetrics class extending mediator, and add metrics code there before and after the original mediator methods. You can return MediatorWithMetrics from LightblueFactory based on config.